providers/gcloud/gcloudProvider.go

```diff
@@ -111,17 +112,56 @@ func New(cfg map[string]string, metadata json.RawMessage) (providers.DNSServiceP
 		oldRRsMap:     map[string]map[key]*gdns.ResourceRecordSet{},
 		zoneNameMap:   map[string]string{},
 	}
+	if len(metadata) != 0 {
+		err := json.Unmarshal(metadata, g)
+		if err != nil {
+			return nil, err
+		}
+		if len(g.Visibility) != 0 {
+			visibilityCheck := regexp.MustCompile("^(public|private)$")
```

Comment:
Would prefer to compile this once at the top

Reply:
moved all of the regexp.MustComplie() calls to var block at the top - hope that was what you were looking for.  let me know if you had something else in mind.